compat/maven-compat/src/test/java/org/apache/maven/repository/legacy/resolver/DefaultArtifactCollectorTest.java

```diff
@@ -293,10 +293,9 @@ void testResolveNearestWithRanges() throws ArtifactResolutionException, InvalidV
     }
 
     @Test
-    @SuppressWarnings("checkstyle:UnusedLocalVariable")
     void testResolveRangeWithManagedVersion() throws ArtifactResolutionException, InvalidVersionSpecificationException {
         ArtifactSpec a = createArtifactSpec("a", "1.0");
-        ArtifactSpec b = a.addDependency("b", "[1.0,3.0]");
```

Comment:
does this [unusedassignment](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedassignment) server a higher purpose?

Reply:
yes, this is unit lvl and can be modified if needed of just give watch. We all extract code into snippets when debugging, but only then. 9 out of 10 its reading the code. Once its running, debug mode is the exception. 

Therefore should be cleaned up to meet out standards.

Same reasoning applies to any code to we dont leave this raw as well so it seems inconsequent imho.

Reply:
thats always the best of course. Thx.

Reply:
Yes , adding an assertNotNull should do the trick.

Reply:
Yes , adding an assertNotNull should do the trick.

Reply:
Actually check style already does the check and specially supressed the warning, so this is not needed imho.

Reply:
added test us give field a purpose.